src/main/java/school/hei/haapi/endpoint/rest/security/SecurityConf.java

```diff
@@ -80,24 +80,26 @@ req, res, null, forbiddenWithRemoteInfo(req))))
         .authenticationProvider(authProvider)
         .addFilterBefore(
             bearerFilter(
-                new NegatedRequestMatcher(
-                    new OrRequestMatcher(
-                        new AntPathRequestMatcher("/ping", GET.name()),
-                        new AntPathRequestMatcher("/uuid-created", GET.name()),
-                        new AntPathRequestMatcher("/health/db", GET.name()),
-                        new AntPathRequestMatcher("/health/email", GET.name()),
-                        new AntPathRequestMatcher("/health/event1", GET.name()),
-                        new AntPathRequestMatcher("/health/event2", GET.name()),
-                        new AntPathRequestMatcher("/health/event/uuids", POST.name()),
-                        new AntPathRequestMatcher("/health/bucket", GET.name()),
-                        new AntPathRequestMatcher("/**", OPTIONS.toString())))),
+                new OrRequestMatcher(
+                    antMatcher(GET, "/whoami"),
+                    antMatcher(GET, "/school/files"),
+                    antMatcher(POST, "/school/files/raw"),
+                    antMatcher(POST, "/students/*/files/raw"),
+                    antMatcher(POST, "/students/*/work_files/raw"),
+                    antMatcher(GET, "/school/files/share_link"),
+                    antMatcher(GET, "/students/*/work_files"),
+                    antMatcher(GET, "/students/*/work_files/*"),
+                    antMatcher(GET, "/students/*/files"),
+                    antMatcher(GET, "/students/*/scholarship_certificate/raw"))),
             AnonymousAuthenticationFilter.class)
 
         // authorize
         .authorizeHttpRequests(
             request ->
                 request
-                    .requestMatchers(GET, "/ping")
+                    .requestMatchers(
+                        new AntPathRequestMatcher("/ping", GET.name()),
+                        new AntPathRequestMatcher("/ping", GET.name()))
```

Comment:
two ping endpoints ?

requestMatchers(GET, "/ping") was already correct

Reply:
you need to write down all ENDPOINTS which require authentication, implying those with .hasAnyRole or .hasRole because before getting someone's role, you have to know their identity (hence authentication)